src/Components/HomePage/ElementsContainer.js

```diff
@@ -1,86 +1,74 @@
-import React, { Suspense } from 'react';
+import React/*, { Suspense } */ from 'react';
+import { Link } from 'react-router-dom';
 
-import { BannerContainer, CategoriesContainer, ProductContainer, SectionHeader } from './styled';
+import { CategoriesContainer, ProductContainer, SectionHeader } from './styled';
 
-const Slider = React.lazy(() => import('../Slider/Slider'));
-const ProductElement = React.lazy(() => import('./ProductElement'));
-const CategoriesElement = React.lazy(() => import('./CategoriesElement'));
-const Button = React.lazy(() => import('Components/Button'));
+import ProductElement from './ProductElement';
+import CategoriesElement from './CategoriesElement';
+import Button from 'Components/Button';
 
-const ElementsContainer = (props) => {
-    //const [fetchedData, setFetchedData] = React.useState(null);
-    
-    //TODO: Check if its really nedded: <Error boundarys />
-    //const [error, setError] = React.useState(null);
+import { useFeaturedBanners } from 'utils/hooks/useFeaturedBanners';
 
-    //TODO: Check if its really needed:  fallback={}
-    //const [isLoading, setIsLoading] = React.useState(false);
-    
-    /* React.useEffect(() => {
-        setIsLoading(true);
-        
-        setError(null);
-        setIsLoading(false);
-    }, []); */
+//const Slider = React.lazy(() => import('../Slider/Slider'));
+//const ProductElement = React.lazy(() => import('./ProductElement'));
+//const CategoriesElement = React.lazy(() => import('./CategoriesElement'));
+//const Button = React.lazy(() => import('Components/Button'));
 
-    const fetchedData = props.source.results;
+const ElementsContainer = ({type, pageSize, optional}) => {
+    const { data, isLoading } = useFeaturedBanners(type, pageSize, optional);
+    /* console.group(type)
+    console.log('Type',type);
+    console.log('Page size', pageSize);
+    console.log('Data', data.results);
+    console.log('Is loading?', isLoading);
+    console.groupEnd(); */
 
-    /* if ( isLoading ){
-        return ('Here should be a skeleton')
-    }
-    else if( error ){
-        return ('Here should be an apologize and a refresh page button')
-    }
-    else  */if( fetchedData.length > 0 ){
-        switch(props.type){
-            case 'Banner' :
-                return(
-                    <Suspense fallback={<div>Loading...</div>}>
-                        <BannerContainer>
-                            <Slider items={fetchedData} />
-                        </BannerContainer>
-                    </Suspense>
-                )
-
-            case 'Categories' :
+    if(!isLoading){
+        switch(type){
+            case 'category' :
```

Comment:
I believe I have added some comments in the previous PRs about these. Please review them once you have time. Otherwise you'll be carrying these over the following deliverables.